schematic/schemas/generator.py

```diff
@@ -38,15 +40,33 @@ def __init__(self,
             None
         """
 
-        # create an instance of SchemaExplorer
-        self.se = SchemaExplorer()
+        if schema_explorer is None:
+
+            assert path_to_json_ld is not None, (
+                "You must provide either `path_to_json_ld` or `schema_explorer`."
+            )
+
+            assert path_to_json_ld.rpartition('.')[-1] == "jsonld", (
+                "Please make sure the 'path_to_json_ld' parameter "
+                "is pointing to a valid JSON-LD file."
+            )
+
+            # create an instance of SchemaExplorer
+            self.se = SchemaExplorer()
 
-        if path_to_json_ld.rpartition('.')[-1] == "jsonld":
             # convert the JSON-LD data model to networkx object
             self.se.load_schema(path_to_json_ld)
+
         else:
-            print("Please make sure the 'path_to_json_ld' parameter is pointing to a valid JSON-LD file.")
-            return
+
+            # Confirm that given SchemaExplorer instance is valid
+            assert getattr(schema_explorer, "schema_nx") is not None, (
+                "SchemaExplorer instance given to `schema_explorer` "
+                "does not have a `schema_nx` attribute."
+            )
+
+            # User given instance of SchemaExplorer
+            self.se = schema_explorer
```

Comment:
In addition to checking for the `schema_explorer.schema_nx` attribute/property, we also need to check if the `schema_explorer.schema` property has been provided. Since `load_schema` sets both those properties, it's necessary that the `schema_explorer` object that the user provides have the same structure as that created by `load_schema`.

Reply:
Good idea! I've addressed this in the faf4062.